src/runtime_src/core/edge/common/aie_parser.cpp

```diff
@@ -208,6 +209,40 @@ get_tiles(const pt::ptree& aie_meta, const std::string& graph_name)
   return tiles;
 }
 
+std::vector<tile_type>
+get_event_tiles(const pt::ptree& aie_meta, const std::string& graph_name,
+                module_type type)
+{
+  // Not supported yet
+  if (type == module_type::shim)
+    return {};
+
+  const char* col_name = (type == module_type::core) ? "core_columns" : "dma_columns";
+  const char* row_name = (type == module_type::core) ?    "core_rows" :    "dma_rows";
+
+  std::vector<tile_type> tiles;
+ 
+  for (auto& graph : aie_meta.get_child("aie_metadata.EventGraphs")) {
+    if (graph.second.get<std::string>("name") != graph_name)
+      continue;
+
+    int count = 0;
+      for (auto& node : graph.second.get_child(col_name)) {
+        tiles.push_back(tile_type());
+        auto& t = tiles.at(count++);
+        t.col = std::stoul(node.second.data());
+      }
+
+      int num_tiles = count;
+      count = 0;
+      for (auto& node : graph.second.get_child(row_name))
+        tiles.at(count++).row = std::stoul(node.second.data());
+      throw_if_error(count < num_tiles,"rows < num_tiles");
```

Comment:
Look at this make me think that `throw_if_error` has not the right name.
`throw_if_true`? `throw_runtime_error_if_true`?

Reply:
Argh...   I'd expect throw_if_error(...) to check the return value of a function (or condition) and flag an error if the value is different from 0.  Alas I think throw_if_error is the right name, you pass it a value that represents the presence of an error.